main.js

```diff
@@ -242,16 +260,11 @@ exports.main = function (options, callbacks) {
 exports.onUnload = function (reason) {
   // If the app is shutting down, skip the rest
   if (reason === 'shutdown') {
-    for (let window of browserWindows) {
-      let win = viewFor(window);
-      let mainWindow = win.document.getElementById('main-window');
-      ss.setWindowValue(win, 'TCtabspinnedwidth', mainWindow.getAttribute('tabspinnedwidth'));
-      ss.setWindowValue(win, 'TCtabspinned', mainWindow.getAttribute('tabspinned'));
-      ss.setWindowValue(win, 'TCtoggledon', mainWindow.getAttribute('toggledon'));
-    }
     return;
   }
-
+  if (startupFinishedObserver) {
+    observerService.removeObserver(startupFinishedObserver, 'browser-delayed-startup-finished');
+  }
```

Comment:
And we probably want to set `startupFinishedObserver` to `null` here, too, just in case…